### PR TITLE
CDRIVER-3241: Package C# libmongocrypt wrapper for driver consumption

### PR DESCRIPTION
**Pull request by vincentkam**
https://jira.mongodb.org/browse/CDRIVER-3241

### CODE REVIEW

**Comment by vincentkam** [comment]
Ready for review!

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj [hunk 1]
```diff
@@ -18,17 +18,17 @@
...
-    <!-- <PackageLicenseFile>License.txt</PackageLicenseFile> -->
+    <PackageLicenseFile>License.txt</PackageLicenseFile>
     <PackageTags>mongodb;mongo;nosql</PackageTags>
     <PackageLanguage>en-US</PackageLanguage>
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
 
   <PropertyGroup>
-    <MDBDeveloperDirectory>../../../build/$(Configuration)</MDBDeveloperDirectory>
+    <MDBDeveloperDirectory>../../../cmake-build/$(Configuration)</MDBDeveloperDirectory>
```
Comment (rstam) [inline bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: I don't know what this property is.

**Comment by vincentkam** [comment]
Sure, let's schedule a Zoom meeting! To address your question briefly though: pushing to MyGet is still a manual process: I have to build the various parts manually and then use the web interface to upload it: it was quicker than trying to create a new script/adapt the driver's CAKE script, although that's certainly a place where we could improve the process!

**Comment by jyemin** [comment]
The implication is that pushing to Nuget will be similarly manual?

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj (hunk 1 above)
Comment (vincentkam) [reply bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: This property specifies the location of where we can find the binaries for `libmongocrypt`. We could probably update the name for clarity if you'd like!

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj [hunk 2]
```diff
@@ -61,6 +61,30 @@
...
   </ItemGroup> -->
+  <ItemGroup>
+    <Content Include="$(MDBDeveloperDirectory)/mongocrypt.dll" >
+      <Pack>true</Pack>
+      <PackagePath>x64/native/windows</PackagePath>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="$(MDBDeveloperDirectory)/mongocrypt.pdb" Condition="'$(Configuration)'=='Debug'">
```
Comment (markbenvenuto) [inline bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: Pdb files are built regardless of release vs debug mode. Does the csharp driver ship pdbs? I would match whatever it does here.

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj (hunk 1 above)
Comment (rstam) [reply bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: Yes please. Perhaps something like `LibMongoCryptBinaries` then?

**Comment by rstam** [comment]
An earlier comment of mine suggested a Zoom meeting. That is no longer necessary. But github does not let me resolve that comment.

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj (hunk 2 above)
Comment (vincentkam) [reply bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: Good catch! The C# driver does ship with PDBs in the .nupkg file, so I agree that the wrapper should do the same. I've also added the `mongocrypt.pdb` file.

Aside: Although I believe the newer way is to create a [.snupkg](https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg) file, even the driver isn't doing this, and due to the time crunch, I'm fine with not going down this route.

File: bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj (hunk 1 above)
Comment (vincentkam) [reply bindings/cs/MongoDB.Crypt/MongoDB.Crypt.csproj]: Done.

**Comment by vincentkam** [comment]
Ready for re-review :)

**Comment by vincentkam** [comment]
Requesting a quick review for the new changes

**Comment by vincentkam** [comment]
Per offline discussion: rebased, cherry picked and merged

**Pull request closed without merging**